Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs

```diff
@@ -2320,3 +2651,648 @@ carouselView state push =
+        , accessibility DISABLE_DESCENDANT
+        , id (getNewIDWithTag "CustomerHomeScreenMap")
+        , visibility if state.props.isSrcServiceable then VISIBLE else GONE
+        , cornerRadius if state.props.currentStage == HomeScreen then 16.0 else 0.0
+        ][]
+     , linearLayout 
+        [ height WRAP_CONTENT
+        , width MATCH_PARENT
+        , alignParentBottom "true,-1"
+        , gravity RIGHT
+        , padding $ Padding 0 0 16 16
+        , visibility $ if isHomeScreenView state then VISIBLE else GONE
+        ][ imageView
+            [ imageWithFallback $ "ny_ic_recenter_btn," <> (getCommonAssetStoreLink FunctionCall) <> "ny_ic_recenter_btn.png"
+            , accessibility DISABLE
+            , onClick
+                ( \action -> do
+                    _ <- push action
+                    _ <- getCurrentPosition push UpdateCurrentLocation
+                    _ <- pure $ logEvent state.data.logField "ny_user_recenter_btn_click"
+                    pure unit
+                )
+                (const $ RecenterCurrentLocation)
+            , height $ V 40
+            , width $ V 40
+            ]
+
+        ]
+    ]
+getMapDimensions :: HomeScreenState -> {height :: Length, width :: Length}
+getMapDimensions state = 
+  let mapHeight = if (any (_ == state.props.currentStage) [RideAccepted, RideStarted, ChatWithDriver ] && os /= "IOS" && not state.props.locateOnMap) then 
+                    V (((screenHeight unit)/ 15)*10)
+                  else if ( (suggestionViewVisibility state) && (isBannerVisible state)|| suggestionViewVisibility state) then 
+                    V (getHeightFromPercent 27)
+                  else if (isHomeScreenView state) then
+                    V (getHeightFromPercent 45)
+                  else
+                    MATCH_PARENT
+      mapWidth =  if state.props.currentStage /= HomeScreen then MATCH_PARENT else V ((screenWidth unit)-32)
+  in {height : mapHeight, width : mapWidth}
+
+suggestionViewVisibility :: HomeScreenState -> Boolean
+suggestionViewVisibility state =  ((length state.data.tripSuggestions  > 0 || length state.data.destinationSuggestions  > 0) && isHomeScreenView state)
+
+isBannerVisible :: HomeScreenState -> Boolean
+isBannerVisible state = getValueToLocalStore DISABILITY_UPDATED == "false" && state.data.config.showDisabilityBanner && isHomeScreenView state
+
+isSaveLocationsVisible :: HomeScreenState -> Boolean
+isSaveLocationsVisible state = true 
+
+suggestionsView :: forall w. (Action -> Effect Unit) -> HomeScreenState -> PrestoDOM (Effect Unit) w
+suggestionsView push state = 
+  linearLayout 
+    [ width MATCH_PARENT
+    , height WRAP_CONTENT
+    , orientation VERTICAL
+    , padding $ Padding 0 0 0 16
+    , margin $ Margin 8 16 8 0
+    ][ textView $
+        [ height WRAP_CONTENT
+        , width (MATCH_PARENT)
+        , text if length state.data.tripSuggestions  == 0 then (getString SUGGESTED_DESTINATION) else (getString RECENT_RIDES)
```

Comment:
instead of having `length state.data.tripSuggestions  == 0` check you can directly use `null` for arrays which returns a boolean value directly if array is null or not